pages/studentAssessmentHomework/studentAssessmentHomework.js

```diff
@@ -47,35 +49,38 @@ router.get('/', function (req, res, next) {
       // If this assessment is group work and there is no existing instance,
       // show the group info page.
       if (res.locals.assessment.group_work) {
-        groupAssessmentHelper.getGroupInfo(
+        // Get the group config info
+        const groupConfig = await groupAssessmentHelper.getGroupConfig(res.locals.assessment.id);
+        res.locals.groupConfig = groupConfig;
+
+        // Check whether the user is currently in a group in the current assessment by trying to get a group_id
+        const groupId = await groupAssessmentHelper.getGroupId(
           res.locals.assessment.id,
-          res.locals.user.user_id,
-          function (
-            err,
-            groupMember,
-            permissions,
-            minsize,
-            maxsize,
-            groupsize,
-            needsize,
-            group_info,
-            join_code,
-            start,
-            used_join_code
-          ) {
-            if (ERR(err, next)) return;
-            res.locals.permissions = permissions;
-            res.locals.minsize = minsize;
-            res.locals.maxsize = maxsize;
-            res.locals.groupsize = groupsize;
-            res.locals.needsize = needsize;
-            res.locals.group_info = group_info;
-            res.locals.join_code = join_code;
-            res.locals.start = start;
-            res.locals.used_join_code = used_join_code;
-            res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
-          }
+          res.locals.user.user_id
         );
+
+        if (groupId === null) {
+          res.locals.notInGroup = true;
+        } else {
+          res.locals.notInGroup = false;
+          const groupInfo = await groupAssessmentHelper.getGroupInfo(groupId, groupConfig);
+          res.locals.groupSize = groupInfo.groupSize;
+          res.locals.groupMembers = groupInfo.groupMembers;
+          res.locals.joinCode = groupInfo.joinCode;
+          res.locals.groupName = groupInfo.groupName;
+          res.locals.start = groupInfo.start;
+          res.locals.rolesInfo = groupInfo.rolesInfo;
+
+          if (groupConfig.has_roles) {
+            const result = await groupAssessmentHelper.getAssessmentLevelPermissions(
```

Comment:
Nit: idk if we need the word "level" here? `getAssessmentPermissions` has the same meaning to me.